go/integration_test/processor/create_test.go

```diff
@@ -0,0 +1,349 @@
+				}...)
+
+				// Add salt for CREATE2
+				if create == vm.CREATE2 {
+					baseCode = append(baseCode, byte(vm.PUSH1), byte(0))
+				}
+
+				// Create the contract
+				baseCode = append(baseCode, []byte{
+					byte(vm.PUSH1), byte(len(initCode)), // input size
+					byte(vm.PUSH1), byte(initCodeOffset), // input offset
+					byte(vm.PUSH1), byte(0), // value
+					byte(create),
+				}...)
+
+				// gas for the call
+				baseCode = append(baseCode, byte(gasPush))
+				baseCode = append(baseCode, gasLimit.Bytes()...)
+
+				// Call contract and return result
+				baseCode = append(baseCode, []byte{
+					byte(vm.CALL),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.SLOAD),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.MSTORE),
+					byte(vm.PUSH1), byte(64),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}...)
+
+				state := WorldState{
+					sender0:               Account{},
+					receiver0:             Account{Code: baseCode},
+					toBeCreatedCodeHolder: Account{Code: codeToBeCreated},
+					initCodeHolder:        Account{Code: initCode},
+				}
+				transaction := tosca.Transaction{
+					Sender:    sender0,
+					Recipient: &receiver0,
+					GasLimit:  sufficientGas,
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+				if err != nil || !result.Success {
+					t.Errorf("execution was not successful or failed with error %v", err)
+				}
+				want := append(bytes.Repeat([]byte{0}, 31), input+increment)
+				want = append(want, append(bytes.Repeat([]byte{0}, 31), otherValue)...)
+				if !slices.Equal(result.Output, want) {
+					t.Errorf("creation of contract or its call was not successful, returned %v", result.Output)
+				}
+			})
+		}
+	}
+}
+
+func TestProcessor_EmptyReceiverCreatesAccount(t *testing.T) {
+	for processorName, processor := range getProcessors() {
+		if strings.Contains(processorName, "floria") {
+			continue // todo implement different create types
+		}
+
+		sender0 := tosca.Address{1}
+		checkValue := byte(42)
+
+		initCode := []byte{
+			byte(vm.PUSH1), checkValue,
+			byte(vm.PUSH1), byte(0),
+			byte(vm.MSTORE),
+			byte(vm.PUSH1), byte(32),
+			byte(vm.PUSH1), byte(0),
+			byte(vm.RETURN),
+		}
+
+		state := WorldState{
+			sender0: Account{},
+		}
+		transaction := tosca.Transaction{
+			Sender:    sender0,
+			Recipient: nil,
+			GasLimit:  sufficientGas,
+			Input:     initCode,
+		}
+
+		transactionContext := newScenarioContext(state)
+
+		// Run the processor
+		result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+		if err != nil || !result.Success {
+			t.Errorf("execution was not successful or failed with error %v", err)
+		}
+		if !slices.Equal(result.Output, append(bytes.Repeat([]byte{0}, 31), checkValue)) {
+			t.Errorf("creation of account was not successful, returned %v", result.Output)
+		}
```

Comment:
The result should contain a `ContractAddress` field that should also be checked here whether it contains the address of the created account.

Reply:
Added it to this integration tests but I mainly want to stress the created account address with unit tests during the implementation in Floria.